linkerd/app/src/env.rs

```diff
@@ -500,6 +507,18 @@ pub fn parse_config<S: Strings>(strings: &S) -> Result<super::Config, EnvError>
     })
 }
 
+fn convert_string_to_map(labels: String) -> HashMap<String, String> {
+    let mut labels_map: HashMap<String, String> = HashMap::new();
+    for line in labels.lines() {
+        let label = line.split("=").collect::<Vec<&str>>();
+        labels_map.insert(
+            label[0].to_string(),
+            label[1][1..label[1].len() - 1].to_string(),
```

Comment:
What does this do? Perhaps break this into multiple statements? Or at least comment what this indexing does... Also, is this guaranteed to be in range or can this cause runtime panics?

Reply:
So, k8s adds double quotes to the label value `cluster="test-cluster1"`. Hence the indexing. 

To make it more generic for cases like mesh expansion, etc. I guess having a check for `"` and performing indexing only when they are present, should be good?

Reply:
What about using [`str::trim_matches`](https://doc.rust-lang.org/stable/std/primitive.str.html#method.trim_matches) instead? Like
```suggestion
            label[1].trim_matches('"').to_string()
```